frontend/src/components/MapView/DateSelector/index.tsx

```diff
@@ -278,16 +376,31 @@ const DateSelector = memo(() => {
 
   // Click on available date to move the pointer
   const clickDate = (index: number) => {
-    const selectedIndex = findDateIndex(availableDates, dateRange[index].value);
+    // Get all dates that are queriable for each layer
+    const queriableDates = truncatedLayers.map(layerDates =>
+      layerDates
+        .filter(dateItem =>
+          datesAreEqualWithoutTime(dateItem.queryDate, dateItem.displayDate),
+        )
+        .map(dateItem => dateItem.displayDate),
+    );
+
+    // Find the dates that are queriable for all layers
+    const validDates = queriableDates.reduce((acc, currentArray) =>
+      acc.filter(date => currentArray.includes(date)),
+    );
+
+    const selectedIndex = findDateIndex(validDates, dateRange[index].value);
+
```

Comment:
This feels like a lot of array looping and gave me some pause. I did measure it though for two HDC layer (with differing validity) and found it takes less than a millisecond on my computer. Not cause for concern imo
<img width="244" alt="Screenshot 2024-07-17 at 10 36 13 AM" src="https://github.com/user-attachments/assets/25e8967b-8ad1-4feb-a191-fb9297cd9079">